blog/models.py

```diff
@@ -0,0 +1,26 @@
+"""Models for the blog app."""
+
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+
+
+class Post(models.Model):
+    """A class to represent a post."""
+
+    author = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE,
+    )
+    title = models.CharField(max_length=200)
+    text = models.TextField()
+    created_date = models.DateTimeField(default=timezone.now)
+    published_date = models.DateTimeField(blank=True, null=True)
+
+    def publish(self):
+        """Save the post entry."""
+        self.published_date = timezone.now
+        self.save()
+
+    def __str__(self):
+        """Return the post title."""
```

Comment:
"post title" is the implementation detail. The caller doesn't know anything about the internals of the post, it just wants it as a string. So the method should provide that string and it's not necessarily a title or some other detail that might change. This will quickly become outdated as you start adding the author and the date and forget to update the additional text.

Since this declares a built-in protocol for turning an object to a string, for this interface I'd use something laong the lines of "Render a Post representation as a string."

Reply:
Thank you! Addressed https://github.com/kpi-web-guild/django-girls-blog-Martolivna/pull/20/commits/c49ed7b948b0f2012c47e3a24d98d3cdf0d3649a.

Reply:
Marking as resolved.